### PR TITLE
boards/arm/stm32f7: separate nucleo-144 board into individual boards

### PR DESCRIPTION
**Pull request by raiden00pl**
## Summary

`nucleo-144` combines 3 different ST boards. This approach is inconsistent with the support for the rest of the nucleo boards in NuttX, where each board is in separate folders. Also `nucleo-144` is no longer reserved for STM32F7 chips but other families also use this format.

After this commit `nucleo-144` is divided into 3 boards:

- nucleo-f746zg
- nucleo-f767zi
- nucleo-f722ze

ROMFS, CAN, SPI test and reset logic have been moved to common logic. 
`stm32_dma_alloc.c` is not used by these ports, so it was removed.

## Impact

Breaking change for users that use nucleo-144 board configuration, they need to migrate to a specific board.

`CONFIG_STM32_ROMFS_xxx` options are now  `CONFIG_STM32F7_ROMFS_xxx` to avoid conflict with stm32 options.

## Testing
building locally new configurations

### CODE REVIEW

**Comment by raiden00pl** [comment]
FYI @robertobucher since you are using `nucleo-144` configurations

**Review by xiaoxiang781216 — APPROVED** [review]

File: boards/arm/stm32f7/nucleo-f767zi/src/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+endif()
+
+if(CONFIG_USBDEV_COMPOSITE)
+  list(APPEND SRCS stm32_composite.c)
+endif()
+
+target_sources(board PRIVATE ${SRCS})
+
+set_property(GLOBAL PROPERTY LD_SCRIPT "${NUTTX_BOARD_DIR}/scripts/flash.ld")
+endif()
```
Comment (lupyuen) [inline boards/arm/stm32f7/nucleo-f767zi/src/CMakeLists.txt:76]: Sorry @raiden00pl I think this `endif()` is missing an `if()?` Here's the error from our Scheduled Merge Job:
```text
Cmake in present: nucleo-f767zi/nsh,CONFIG_ARM_TOOLCHAIN_CLANG
CMake Error at boards/arm/stm32f7/nucleo-f767zi/src/CMakeLists.txt:76 (endif):
  Flow control statements are not properly nested.
```text
https://github.com/nuttxpr/nuttx/actions/runs/11439627928/job/31823813226
```

Comment (lupyuen) [reply boards/arm/stm32f7/nucleo-f767zi/src/CMakeLists.txt:76]: Seems to be fixed here: https://github.com/apache/nuttx/pull/14437

**Comment by raiden00pl** [comment]
another fix is here https://github.com/apache/nuttx/pull/14439

**Pull request merged**